charms/worker/k8s/src/charm.py

```diff
@@ -137,9 +135,7 @@ def _create_cluster_tokens(self):
             return
 
         if peer := self.model.get_relation("cluster"):
-            self._distribute_cluster_tokens(peer, "control-plane")
-
-        # TODO handle requesting cluster tokens for workers
+            self._distribute_cluster_tokens(peer)
```

Comment:
I think this needs to be called again with the app relation:


```suggestion
            self._distribute_cluster_tokens(peer)

        if workers := self.model.get_relation("k8s-cluster"):
            self._distribute_cluster_tokens(workers)
```